hydromt/data_catalog.py

```diff
@@ -1594,6 +1595,85 @@ def get_geodataset(
         )
         return obj
 
+    def get_dataset(
+        self,
+        data_like: str | SourceSpecDict | Path | xr.Dataset | xr.DataArray,
+        variables: Optional[List] = None,
+        time_tuple: Optional[Tuple[str] | Tuple[datetime]] = None,
+        single_var_as_array: bool = True,
+        provider: Optional[str] = None,
+        version: Optional[str] = None,
+        **kwargs,
+    ) -> xr.Dataset:
+        """Return a clipped, sliced and unified Dataset.
+
+        To slice the data to the time period of interest, provide the
+        `time_tuple` argument. To return only the dataset variables
+        of interest provide the `variables` argument.
+
+        NOTE: Unless `single_var_as_array` is set to False a single-variable data source
+        will be returned as xarray.DataArray rather than Dataset.
```

Comment:
Especially now we need to start being a little more careful with making clear whether we mean a HydroMT dataset or an xarray dataset.